loaders/src/glTF/2.0/Extensions/KHR_materials_variants.ts

```diff
@@ -0,0 +1,131 @@
+    if (variantName instanceof Array) {
+      variantName.forEach((name) => this.selectVariantTag(name));
+    } else {
+      this.selectVariantTag(variantName);
+    }
+  }
+
+  public selectVariantTag(variantName: string) {
+    // If the name is valid, switch all meshes to use materials defined by the tags
+    const variantMappings = this._tagsToMap.get(variantName);
+    if (variantMappings === undefined) {
+      return;
+    }
+    variantMappings.forEach((mapping: VariantMapping) => {
+        if (mapping.material) {
+            mapping.mesh.material = mapping.material;
+            return;
+        }
+        mapping.materialPromise.then((material) => {
+            mapping.mesh.material = material;
+        });
+    });
+  }
+
+  /** @hidden */
+  public onLoading(): void {
+    const extensions = this._loader.gltf.extensions;
+    if (extensions && extensions[this.name]) {
+        const extension = extensions[this.name] as IKHRMaterialVariantsTop;
+        this.defaultVariant = extension.default;
+    }
+  }
+
+  public _loadMeshPrimitiveAsync(context: string, name: string, node: INode, mesh: IMesh, primitive: IMeshPrimitive, assign: (babylonMesh: AbstractMesh) => void): Promise<AbstractMesh> {
+    const loadExtensions = GLTFLoader.LoadExtensionAsync<IKHRMaterialVariants, AbstractMesh>(context, primitive, this.name, (extensionContext, extension) => {
+      
+      const assignMesh = (babylonMesh: AbstractMesh) => {
+        assign(babylonMesh);
+        const babylonDrawMode = (GLTFLoader as any)._GetDrawMode(context, primitive.mode);
+        // For each mapping, look at the tags and make a new entry for them
+        extension.mapping.forEach((mapping: IKHRMaterialVariantsMapping) => {
+          mapping.tags.forEach((tag: string, index: number) => {
+            let tag_mapping = this._tagsToMap.get(tag);
+            if (tag_mapping === undefined) {
+                tag_mapping = [];
+            }
+            const material = ArrayItem.Get(`#/materials/`, this._loader.gltf.materials, mapping.material);
+            let meshEntry: VariantMapping = {
+              mesh: babylonMesh,
+              materialPromise: Promise.resolve(null)
+            };
+            if (babylonMesh instanceof Mesh) {
+              const matPromise = this._loader._loadMaterialAsync(`#/materials/${mapping.material}`, material, babylonMesh, babylonDrawMode, (material) => {
+                meshEntry.material = material;
+              });
+              meshEntry.materialPromise = matPromise;
+            }
+            tag_mapping.push(meshEntry);
+            this._tagsToMap.set(tag, tag_mapping);
+          });
+        });
+      };
+      return this._loader._loadMeshPrimitiveAsync(context, name, node, mesh, primitive, assignMesh);
+    });
+
+    if (loadExtensions) {
+        return loadExtensions;
+    }
+    return this._loader._loadMeshPrimitiveAsync(context, name, node, mesh, primitive, assign);
```

Comment:
That's what I originally did but I was getting a TS error because `GLTFLoader.LoadExtensionAsync`'s return value is nullable and `_loadMeshPrimitiveAsync` isn't.
```'Nullable<Promise<AbstractMesh>>' is not assignable to type 'Promise<AbstractMesh>'```

Reply:
Fixed base _loadMeshPrimitiveAsync function to be nullable.